File: Data Structures & Algorithms/products-of-array-discluding-self/submission-4.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        p = 1
        result = []
        for n in nums:
            result.append(p)
            p *= n
        p = 1
        for i in range(len(nums)-1, -1, -1):
            result[i] = result[i] * p
            p *= nums[i]
        return result

File: Data Structures & Algorithms/two-integer-sum/submission-4.go
func twoSum(nums []int, target int) []int {
    complement := make(map[int]int)

	for i, n := range nums {
		if j, ok := complement[target-n]; ok {
			return []int{j, i}
		}
		complement[n] = i
	}
	return nil
}
